dogfood/Dockerfile

```diff
@@ -210,7 +210,7 @@ RUN systemctl enable \
 ARG CLOUD_SQL_PROXY_VERSION=1.26.0 \
 	DIVE_VERSION=0.10.0 \
 	DOCKER_GCR_VERSION=2.1.0 \
-	GOLANGCI_LINT_VERSION=1.48.0 \
+	GOLANGCI_LINT_VERSION=1.50.1 \
```

Comment:
When I was using the go 1.20rc, golangci-lint v1.50.1 lead to massive memory consumption. I noticed CI failed, this could be the reason (OOM killed).

Reply:
Seems this might be fixed in `v1.51.0`, if we switch to that.